### PR TITLE
fix: achieve 100% opset compliance across all supported ONNX operators

### PR DESCRIPTION
**Pull request by antimora**
## Summary

- Lower `min_opset` for all supported ONNX operators to their earliest compatible opset version
- Add attribute extraction fallbacks for operators that migrated from attributes to inputs across opset versions (Slice, Split, Pad, Tile, Resize, BatchNormalization, Softmax, LogSoftmax, Hardmax)
- Achieve **461/461 passing** opset compliance tests (100%), up from ~300 passing before this PR

## Changes by category

**Simple unary/binary ops (min_opset -> 1):** Abs, Neg, Ceil, Floor, Sqrt, Exp, Log, Reciprocal, Relu, Sigmoid, Tanh, Softplus, Softsign, HardSigmoid, LeakyRelu, Selu, PRelu, InstanceNorm, Add, Sub, Mul, Div, Equal, Greater, Less, Gemm, Mean, Sum, Clip, Concat, Reshape, Squeeze, Unsqueeze, Reduce* (10 ops), ArgMax, ArgMin, TopK, Split, Slice, Pad, Tile, BatchNormalization, Softmax, LogSoftmax, Hardmax, AveragePool

**Attribute-to-input migrations:**
- Slice: extract starts/ends/axes from attributes for opset <10
- Split: extract split sizes from attribute for opset <13
- Pad: handle `"paddings"` attribute name (opset 1) vs `"pads"` (opset 2+)
- Tile: extract `"repeats"` from attribute for opset <6
- Resize: opset-dependent input layout (`[X, scales]` in opset 10 vs `[X, roi, scales, sizes]` in 11+)

**Opset-aware behavior:**
- Softmax/LogSoftmax/Hardmax: axis default is 1 for opset <13, -1 for opset 13+
- BatchNormalization: `OutputSpec::Range(1,5)` to handle 5 required outputs in opset <9; ignore deprecated `spatial`/`consumed_inputs` attributes
- Resize: default `coordinate_transformation_mode` to `"asymmetric"` for opset 10

## Test plan

- [x] All 461 opset compliance tests pass (0 failures)
- [x] All onnx-ir unit tests pass
- [x] All burn-onnx tests pass
- [x] All 465 onnx-tests integration tests pass

### CODE REVIEW

File: crates/onnx-ir/src/node/tile.rs [hunk 1]
```diff
@@ -107,22 +107,28 @@ impl NodeProcessor for TileProcessor {
...
-                TileInput::Static(vec![])
             }
+            // Check for repeats attribute (opset 1-5)
+            if let Some(attr) = node.attrs.get("repeats") {
+                let values = attr.clone().into_i64s();
+                let repeats = values.iter().map(|&x| x as usize).collect();
+                return TileInput::Static(repeats);
+            }
+            // No repeats provided - default to empty
+            TileInput::Static(vec![])
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/tile.rs]: Valid point. Tile always had `repeats` as a required input across all opset versions. The attribute fallback was incorrect.

Fixed in 7170e76c: reverted `extract_config` to always read repeats from input (no attribute path), and the test generator now always produces Tile with the `repeats` tensor input for all opsets.

Note: `InputSpec::AtLeast(1)` is kept because the pipeline needs to handle optional/missing inputs gracefully during parsing, but the operator semantics correctly require 2 inputs.

File: crates/onnx-ir/tests/scripts/gen_opset_compliance.py [hunk 2]
```diff
@@ -456,9 +456,14 @@ def make_slice_op(op_name: str, opset: int):
...
-    node = helper.make_node(op_name, [_p(op_name, "input"), _p(op_name, "repeats")], [_p(op_name, "output")], name=_p(op_name, "node"))
-    return [node], [inp], [out], [repeats]
+    if opset >= 6:
+        repeats = numpy_helper.from_array(np.array([2, 3], dtype=np.int64), name=_p(op_name, "repeats"))
+        node = helper.make_node(op_name, [_p(op_name, "input"), _p(op_name, "repeats")], [_p(op_name, "output")], name=_p(op_name, "node"))
+        return [node], [inp], [out], [repeats]
+    else:
+        # Opset 1: repeats is an attribute (tiles/repeats)
+        node = helper.make_node(op_name, [_p(op_name, "input")], [_p(op_name, "output")], name=_p(op_name, "node"), repeats=[2, 3])
+        return [node], [inp], [out], []
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/tests/scripts/gen_opset_compliance.py]: Agreed. Fixed in 7170e76c: `make_tile` now always generates Tile with the `repeats` tensor input for all opset versions, matching the spec.

File: crates/onnx-ir/src/node/split.rs [hunk 3]
```diff
@@ -328,6 +335,34 @@ impl NodeProcessor for SplitProcessor {
                     }
                 }
             };
+        } else if let Some(split_attr) = node.attrs.get("split") {
+            // For opset < 13, split sizes are an attribute
+            let sizes = split_attr.clone().into_i64s();
+            if !sizes.is_empty() {
+                let usizes: Vec<usize> = sizes.into_iter().map(|s| s as usize).collect();
+
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/split.rs:348]: Good catch. Fixed in 7170e76c: added validation that rejects negative split sizes before the `as usize` conversion, matching the input-based validation path.

File: crates/onnx-ir/src/node/reshape.rs [hunk 4]
```diff
@@ -258,9 +264,9 @@ impl NodeProcessor for ReshapeProcessor {
 
     fn spec(&self) -> NodeSpec {
         NodeSpec {
-            min_opset: 5,
+            min_opset: 1,
             max_opset: None,
-            inputs: InputSpec::Exact(2),
+            inputs: InputSpec::AtLeast(1),
             outputs: OutputSpec::Exact(1),
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/reshape.rs:270]: The spec file in this repo (`onnx-spec/ops/Reshape.md`) only reflects the latest opset version. Looking at the actual ONNX spec history:

- **Opset 1-4**: `shape` was an attribute (`AttrType.INTS`), input was just the data tensor (1 input)
- **Opset 5+**: `shape` moved from attribute to a required second input (2 inputs)

This attribute-to-input migration is a well-known ONNX pattern (same as Split, Pad, etc.). The `gen_opset_compliance.py` script correctly generates attribute-based Reshape for opset <5, and these models parse and pass tests successfully.

No changes needed here.
